phpBB/phpbb/auth/provider/oauth/oauth.php

```diff
@@ -163,7 +163,7 @@ public function login($username, $password)
 		$provider = $this->request->variable('oauth_service', '', false);
 		$service_name = $this->get_service_name($provider);
 
-		if ($provider === '' || !array_key_exists($service_name, $this->service_providers))
+		if ($provider === '' || !array_key_exists($service_name, (array) $this->service_providers))
```

Comment:
My preference would probably be `offsetExists($service_name)` here.

Reply:
> 
> 
> My preference would probably be `offsetExists($service_name)` here.

Why?

Reply:
Because the service collection implement ArrayAccess and therefore the default goto function should be https://www.php.net/manual/en/arrayaccess.offsetexists.php

Reply:
You also make the dependency on ArrayAccess explicit. Your code depends on it by using the array cast (if the class wouldn't derive from ArrayAccess, the cast would return class properties). Essentially, by making the dependency explicit, this should probably break tests if the service collection class were to change. Which is probably what we would want to happen.